src/types.ts

```diff
@@ -1,8 +1,21 @@
 import * as vscode from "vscode";
+import type { Version } from "./version";
 
 type Dict<T> = { [x: string]: T };
-export type Tool = { path: string; version: [number, number, number] };
-export type ToolCheckFunc = () => Promise<{ tool?: Tool; error?: string }>;
+
+export type Tool = { path: string; version: Version };
+
+export interface ToolCheckSuccessResult {
+  tool: Tool;
+  error?: undefined;
+}
+export interface ToolCheckErrorResult {
+  tool?: undefined;
+  error: string;
+}
```

Comment:
why an interface vs a type?

Reply:
Just out of habit, normally my linter / the linter settings complain on object types, so I already write them this way, but if this codebase prefers `type xy = {...}` I'll chnage it

Reply:
I changed it to use `type`, the linter accepts it, and literally one line above also type is used xD